dvc/remote/base.py

```diff
@@ -256,7 +256,7 @@ def _get_dir_info_checksum(self, dir_info):
 
         from_info = PathInfo(tmp)
         to_info = self.cache.path_info / tmp_fname("")
-        self.cache.upload(from_info, to_info, no_progress_bar=True)
+        self.cache.upload(from_info, to_info, no_progress_bar=False)
```

Comment:
@efiop wanted to ask if this change is OK

P.S.: ignore the other changes in this PR for now (they'll disappear after #3316 is merged).

Reply:
just to add to this - if this change is OK then there's a lot of nice tidying/refactoring possible since `upload()` would never require a `no_progress_bar` argument

Reply:
@casperdcl Btw, how does the pbar look for this operation? I'm worried about it having some random looking paths in it.

Reply:
this would occur on `dvc add <dir>`. It gets proper file names and is cleared away upon completion. Padding it with some persistent logging messages (and padding `fs.copyfile` with some logging & `sleep(5)`) gives this:

[![asciicast](https://asciinema.org/a/a9dhhofKVQbElIJhwr78T4WgZ.svg)](https://asciinema.org/a/a9dhhofKVQbElIJhwr78T4WgZ)

Reply:
@casperdcl Yeah, not sure about that. That random name is not very helpful. How about we put some desc instead? There is a `name` param for that `upload` method, so should be pretty simple. Though, not sure how to describe this in a user-friendly way. Btw, did this particular part of code cause any issues in your research? It works with tiny files and usually uploading them is very quick.

As to the name here, maybe something like "Computing effective hash for directory '{}'" would be good enough.

Reply:
Hmm. I was interested in adding this just because it would then mean we can drop `no_progress_bar` from **all** `upload()` functions. I don't think it's really useful to have a progress bar for this `cache.upload()` function though so maybe we should leave it for now.

Reply:
@casperdcl So it doesn't make dvc hang? We need to do research when doing such things.

Reply:
No I haven't found any hanging. In any case there's a master progress bar above this line which increments when `cache.upload()` completes.

It's possible if in future we add a "don't display progress bar until 4 seconds have elapsed" https://github.com/tqdm/tqdm/issues/704 then it may be useful to enable progress for `cache.upload()` for very large files.

Reply:
@casperdcl Got it. So the plan is to only refactor the defaults in those remotes, right?

Reply:
If we allow `no_progress_bar=False` then we can refactor away. But perhaps for now we should keep `no_progress_bar=True` and not refactor.

Reply:
@casperdcl Sorry, looks like I'm not following. So gdrive/gs had `no_progress_bar=True`, which contradicts the other remotes. So we will fix those, right? Or are we going to do something else as well? :slightly_smiling_face: Just trying to understand the plan for this PR.

Reply:
I was (and have always been) only talking about this specific `cache.upload()` line. The PR is now fine to merge.

`gdrive`/`gs` things elsewhere in this PR are no-brainers I think.

Reply:
Just to clarify further - if we change the `cache.upload()` line to say `no_progress_bar=False`, then it would mean every single instance of `upload()` in the code base would only ever have `no_progress_bar=False` and thus we can drop that argument from all `upload()` functions.